torch/csrc/utils/pybind.h

```diff
@@ -21,7 +21,7 @@ template <> struct type_caster<at::Tensor> {
   bool load(handle src, bool) {
     PyObject* obj = src.ptr();
     if (THPVariable_Check(obj)) {
-      value = ((THPVariable*)obj)->cdata.data();
```

Comment:
Yeah, I was surprised when I read this code, because it is actually _wrong_. Unwrapping a variable along the python<->c++ boundary is not an accurate conversion of the same object into C++. Wherever this is used needs to be updated to unwrap the variables manually.

Reply:
I spoke to Sam about this, and we are going to fix these conversions when Tensor/Variable get merged and the semantics are more clear. For now, I reverted this change and modified to wrappers to work around it.